Add view_count scorer to semantic search backend

appleby opened the pull request:
Closes BOT-311

### Description

Add the `view_count` metadata scorer to the semantic search backend. The implementation largely follows the one for the appdb backend, but updated to query the semantic index rather than the appdb to get the `view-count-percentiles`.

### How to verify

Search for some items that have been viewed, and verify they get a non-zero `view-count` component in their score. Note the item has to be reindexed after viewing it in order for the `view_count` to be recorded in the index.

```
{
    "score": 1.0,
    "name": "view-count",
    "weight": 2,
    "contribution": 2.0
}
```

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

appleby committed:
Add view_count scorer to semantic search backend

bronsa commented on enterprise/backend/src/metabase_enterprise/semantic_search/scoring.clj:
```diff
@@ -14,6 +19,41 @@
+       :arglists '([db index-table p-value])}
+  view-count-percentiles
+  (if config/is-prod?
+    (memoize/ttl view-count-percentiles*
```
sticking `db` in a memoize table makes me a bit uneasy. I guess for now it's fine as it'd be a singleton datasource, but if we ever end up passing a Connection we may have object leaks which could grow even in 1h

appleby replied:
I was a bit worried about this as well tbh. What do you think about ignoring the db arg for caching purposes since it "should" never change? Obviously, that has bad edge cases of its own if it *does* change. Maybe `db` shouldn't be an argument to this function and it should just directly use `get-pgvector-datasource!` to enforce using the singleton connection?

bronsa replied:
yeah I think that's better

appleby committed:
Get pgvector db from global env in semantic.scoring/view-count-percentiles*

Get the db data-source directly rather than passing it in as an argument to this function to side-step potential
issues with using the db as a cache key for the memoized version.

https://github.com/metabase/metabase/pull/62086#discussion_r2272790618
